File: sgilt-core/src/main/java/net/franzka/sgilt/core/prestataire/dto/PrestataireDetailDto.java
package net.franzka.sgilt.core.prestataire.dto;

import net.franzka.sgilt.core.prestataire.domain.Engagement;
import net.franzka.sgilt.core.prestataire.domain.PrestataireStatus;

import java.util.List;

/**
 * DTO complet pour la fiche détail d'un prestataire.
 */
public record PrestataireDetailDto(
        String id,
        String name,
        String slug,
        String baseline,
        String avatar,
        String shortDescription,
        String categoryKey,
        List<String> subcatKeys,
        List<MediaDto> medias,
        List<Engagement> badges,
        List<String> offerings,
        IdentityDto identity,
        String budget,
        List<TestimonialDto> testimonials,
        List<DetailDto> details,
        List<FaqItemDto> faq,
        PrestataireStatus status
) {}

File: sgilt-front/app/data/prestataire/dto/PrestataireDetailDto.ts
/**
 * DTO — contrat API pour la fiche complète d'un prestataire
 */
import type { EngagementKey } from '~/utils/constants'
import type { Media } from '../domain/Media'

export interface PrestataireDetailDto {
  id: string
  name: string
  shortDescription: string
  slug: string
  categoryKey: string
  baseline: string
  avatar: string | null
  subcatKeys: string[]
  medias: Media[]
  badges: EngagementKey[]
  offerings: string[]
  identity?: { quote: string; bio: string }
  budget: string | null
  testimonials?: { author: string; text: string; eventType?: string }[]
  logistics?: string[]
  technical?: string[]
  faq?: { question: string; answer: string }[]
}
